### PR TITLE
Type "DELETE" resource delete confirmation message

### PR DESCRIPTION
**Pull request by sblack-usu**
#fixes 3890

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. Press the delete icon on a resource you have owner permissions for.  Confirm the Delete button is disabled until you type "DELETE" in the textbox.
2. Ensure the textbox is cleared when the dialog is closed and reopened.
3. Ensure the resource is deleted when delete is clicked.
4. Follow the same steps on the my-resources page.

### CODE REVIEW

File: theme/static/js/generic-resource.js [hunk 1]
```diff
@@ -338,9 +338,10 @@ $(document).ready(function () {
         // disable the delete icon to prevent users from clicking it again until it is done
         $('#delete').addClass('disabled');
         $('#delete').removeAttr('data-toggle');
+        let deleteText = $('#confirm-res-id-text').val();
         $.ajax({
             type: "POST",
-            url: "/hsapi/_internal/" + SHORT_ID + "/delete-resource/",
+            url: "/hsapi/_internal/" + SHORT_ID + "/delete-resource/" + "deleteText" + "/",
```
Comment (hyi) [inline theme/static/js/generic-resource.js:344]: deleteText is a variable here, so why quota it as a string in the url input? Should it be just ```+ deleteText``` instead?

Comment (sblack-usu) [reply theme/static/js/generic-resource.js]: That was a typo.

**Review by hyi — DISMISSED** [review]

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|
| NEW SECURITY HOTSPOTS | 0 |
| NEW CODE SMELLS | 0 |
| NEW BUGS | 0 |
| NEW DUPLICATED LINES DENSITY | 0.0 |
| NEW VULNERABILITIES | 0 |

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4313](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4313)

**Comment by sblack-usu** [comment]
retest this, please

**Review by hyi — DISMISSED** [review]

**Review by hyi — APPROVED** [review]

**Pull request merged**